src/components/input/NumericInput.tsx

```diff
@@ -176,6 +182,14 @@ const NumericInput: GenericComponent<NumericInputProps> = forwardRef<HTMLElement
     };
 
     const handleDecreaseMouseDown = () => {
+      if (onDecreaseMouseDown) {
+        onDecreaseMouseDown();
+
+        if (isMinReached()) {
+          return;
+        }
```

Comment:
Q: why does this only need to happen if `onDecreaseMouseDown` is passed?

Reply:
Good question. It's because in all other cases, this is already covered by the buttons becoming disabled and the min/max boundaries present in `updateStep`. Only if we hook into this by passing `onDecreaseMouseDown`, and the min/max has been reached, we should no longer do an increase or decrease, so the return is mainly there to prevent the rest of the function being executed.